packages/rust/server/src/models/group.rs

```diff
@@ -0,0 +1,224 @@
+                .bind(&self.group_name)
+                .fetch_one(&mut tx)
+                .await?;
+        self.id = row.get("id");
+        self.created = row.get("created");
+        sqlx::query(
+            "INSERT INTO user_group (user_id, group_id, role_id)
+                    VALUES ($1, $2, $3);",
+        )
+        .bind(user_id)
+        .bind(self.id)
+        .bind(Role::Admin as i32)
+        .execute(&mut tx)
+        .await?;
+        tx.commit().await?;
+        Ok(())
+    }
+
+    pub async fn update(&self, db: &DbPool) -> Result<(), sqlx::Error> {
+        sqlx::query("UPDATE \"group\" SET group_name = $1 WHERE id = $2")
+            .bind(&self.group_name)
+            .bind(self.id)
+            .execute(db)
+            .await?;
+        Ok(())
+    }
+
+    pub async fn delete(&self, db: &DbPool) -> Result<(), sqlx::Error> {
+        let mut tx = db.begin().await?;
+        sqlx::query("DELETE FROM user_group WHERE group_id = $1")
+            .bind(self.id)
+            .execute(&mut tx)
+            .await?;
+        sqlx::query("DELETE FROM location_group WHERE group_id = $1")
+            .bind(self.id)
+            .execute(&mut tx)
+            .await?;
+        sqlx::query("DELETE FROM \"group\" WHERE id = $1")
+            .bind(self.id)
+            .execute(&mut tx)
+            .await?;
+        tx.commit().await?;
+        Ok(())
+    }
+
+    pub async fn add_user(&mut self, db: &DbPool, user: User) -> Result<(), sqlx::Error> {
+        sqlx::query(
+            "INSERT INTO user_group (user_id, group_id, role_id)
+                    VALUES ($1, $2, $3);",
+        )
+        .bind(user.id)
+        .bind(self.id)
+        .bind(Role::Member as i32)
+        .execute(db)
+        .await?;
+        self.members.push(user.uuid);
+        Ok(())
+    }
+
+    pub async fn delete_user_by_user_id(db: &DbPool, user_id: i32) -> Result<(), sqlx::Error> {
+        let mut tx = db.begin().await?;
+
+        let rows = sqlx::query("DELETE FROM user_group WHERE user_id = $1 RETURNING *;")
+            .bind(user_id)
+            .fetch_all(&mut tx)
+            .await?;
+        for row in rows {
+            let group_id: i32 = row.get("group_id");
+            let groups = sqlx::query("SELECT * FROM user_group WHERE group_id = $1;")
+                .bind(group_id)
+                .fetch_all(&mut tx)
+                .await?;
+            if groups.is_empty() {
+                sqlx::query("DELETE * FROM location_group WHERE group_id = $1;")
+                    .bind(group_id)
+                    .execute(&mut tx)
+                    .await?;
+                sqlx::query("DELETE * FROM \"group\" id = $1;")
+                    .bind(group_id)
+                    .execute(&mut tx)
+                    .await?;
+            }
+        }
+        tx.commit().await?;
+        Ok(())
+    }
+
+    pub async fn delete_user(&mut self, db: &DbPool, user: User) -> Result<(), sqlx::Error> {
```

Comment:
shouldn't we check that the group isn't empty?

Reply:
fixed. a group will always have at least on user. if there is only one user, that user must be an admin. there must be at least on admin.